BlueprintUI/Sources/Layout/Builder/Builder.swift

```diff
@@ -0,0 +1,35 @@
+/// Generic result builder for converting blocks of `Child...` into `[Child]`.
+@_functionBuilder // change to @resultBuilder when we upgrade to Swift 5.4.
+public struct ElementBuilder {
+    /// Function builder for converting blocks of `Child...` into `[Child]`.
+    /// - Parameter children: All children.
+    /// - Returns: `[Child]`.
+    public static func buildBlock(_ elements: Element...) -> [Element] {
+        elements.map { element -> [Element] in
+            if let many = element as? ElementContainer {
+                return many.elements
+            } else {
+                return [element]
+            }
+        }.flatMap { $0 }
+    }
+
+    public static func buildOptional(_ elements: [Element]?) -> Element {
+        ElementContainer(elements: elements ?? [])
+    }
+
+    public static func buildEither(first: [Element]) -> Element {
+        ElementContainer(elements: first)
+    }
+
+    public static func buildEither(second: [Element]) -> Element {
+        ElementContainer(elements: second)
+    }
+}
+
+private struct ElementContainer: ProxyElement {
```

Comment:
@kyleve I'm curious to hear your thoughts here. The function builder needs the types to be homogenous so it doesn't allow for me to mix arrays of Elements and single Elements . To solve for it I created this private wrapper Element that should never render and simply holds an array of elements. Then in `buildBlock` I check to see if the element can be cast as an ElementContainer and if so I extract out the contained elements. I got the idea for ElementContainer from SwiftUI's ForEach but there those views can actually be rendered and have uses beyond what I am doing here.

Reply:
Never mind here....I was able to get rid of it. Just needed the weekend to clear my head 😄